rest-service/manager_rest/storage/resource_models.py

```diff
@@ -656,18 +661,123 @@ def _get_identifier_dict(self):
         id_dict['status'] = self.status
         return id_dict
 
-    def set_deployment(self, deployment, blueprint_id=None):
-        self._set_parent(deployment)
-        self.deployment = deployment
-        current_blueprint_id = blueprint_id or deployment.blueprint_id
-        self.blueprint_id = current_blueprint_id
-
     @classproperty
     def resource_fields(cls):
         fields = super(Execution, cls).resource_fields
         fields.pop('token')
         return fields
 
+    @validates('deployment')
+    def _set_deployment(self, key, deployment):
+        self._set_parent(deployment)
+        if self.blueprint_id is None:
+            self.blueprint_id = deployment.blueprint_id
+        if self.parameters is not None and self.workflow_id is not None:
+            self.merge_workflow_parameters(
+                self.parameters, deployment, self.workflow_id)
+        return deployment
+
+    @validates('workflow_id')
+    def _validate_workflow_id(self, key, workflow_id):
+        if self.parameters is not None and self.deployment is not None:
+            self.merge_workflow_parameters(
+                self.parameters, self.deployment, workflow_id)
+        return workflow_id
+
+    @validates('parameters')
+    def _validate_parameters(self, key, parameters):
+        parameters = parameters or {}
+        if self.workflow_id is not None and self.deployment is not None:
```

Comment:
I'm wondering if all of these checks wrapped around merge_workflow_parameters could just be rolled into merge_workflow_parameters, which might even remove the need for this _validate_parameters function entirely? (I'm not certain about that bit, haven't checked the decorator)

Reply:
the point is, i want the main `merge` method to run only when all three are set. And these one-param-validate methods run in an indeterminate order. I tried toying with writing something more generic but I found that this was the cleanest after all

Reply:
and, in the validate methods, the just-set parameter is passed in as an argument (not on `self` yet) while the others are possibly already set on `self`

Reply:
moving it into the main method sounds like a plan but when I wrote it, it doesn't even save a line, so I'll just not do it instead